tests/test_audit_settings.py

```diff
@@ -0,0 +1,110 @@
+import typer
+from orgwarden.repository import Repository
+from orgwarden.audit_settings import (
+    RepositorySettings,
+    parse_settings_string,
+    append_settings,
+)
+import pytest
+from pytest import CaptureFixture
+
+
+class TestParseSettingsString:
+    def test_invalid_settings_strings(self):
+        INVALID_STRINGS = [
+            "",
+            "just_a_key",
+            "key_no_value: ",
+            ": value_no_key",
+            ":",
+        ]
+        for settings_string in INVALID_STRINGS:
+            with pytest.raises(typer.BadParameter):
+                _ = parse_settings_string(settings_string)
+
+    def test_valid_settings_strings(self):
+        TEST_CASES = [  # input, expected
+            ("key: value", RepositorySettings("key", "value")),
+            ("key: multiple values", RepositorySettings("key", "multiple values")),
+            ("repo: --arg-1 --arg-2", RepositorySettings("repo", "--arg-1 --arg-2")),
+            ("repo    :    value", RepositorySettings("repo", "value")),
+            ("too: many: colons", RepositorySettings("too", "many: colons")),
+            (
+                "'quoted_key': 'quoted_val'",
+                RepositorySettings("'quoted_key'", "'quoted_val'"),
+            ),
+            (
+                '"quoted_key": "quoted_val"',
+                RepositorySettings('"quoted_key"', '"quoted_val"'),
+            ),
+            (
+                "repo: --arg-1     --arg-2",
+                RepositorySettings("repo", "--arg-1     --arg-2"),
+            ),
+        ]
+        for settings_string, expected in TEST_CASES:
+            actual = parse_settings_string(settings_string)
+            assert actual == expected
+
+
+class TestAppendSettings:
+    def test_no_mutation(self):
+        REPOS = [
+            Repository(name="repo_1", url="url_1", org="org_1"),
+            Repository(name="repo_2", url="url_2", org="org_2"),
+            Repository(name="repo_3", url="url_3", org="org_3"),
+        ]
+        new_repos = append_settings(REPOS, [])
+        for new_repo, og_repo in zip(new_repos, REPOS):
+            assert new_repo is not og_repo
+
+    def test_raises_with_duplicate_repos(self):
+        REPOSITORY_SETTINGS = [
+            RepositorySettings("repo", "value 1"),
+            RepositorySettings("repo", "value 2"),
+        ]
+        with pytest.raises(ValueError):
```

Comment:
It's generally a good idea to test not only the type of exception, but also the text of the exception as well. This can be useful if the called function can raise `ValueError` for a variety of different reasons.

Consider:

```
with pytest.raises(ValueError, "I expect this string to be in the error message"):
```